include/zephyr/arch/riscv/arch_inlines.h

```diff
@@ -10,24 +10,20 @@
 #ifndef _ASMLANGUAGE
 
 #include <zephyr/kernel_structs.h>
+#include "csr.h"
 
 static ALWAYS_INLINE uint32_t arch_proc_id(void)
 {
-	uint32_t hartid;
-
-#ifdef CONFIG_SMP
-	__asm__ volatile("csrr %0, mhartid" : "=r" (hartid));
-#else
-	hartid = 0;
-#endif
-
-	return hartid;
+	return csr_read(mhartid);
 }
 
 static ALWAYS_INLINE _cpu_t *arch_curr_cpu(void)
 {
-	/* linear hartid enumeration space assumed */
-	return &_kernel.cpus[arch_proc_id()];
+#if defined(CONFIG_SMP) || defined(CONFIG_USERSPACE)
```

Comment:
I think this might present a problem, as shown tests/kernel/mp 
where CONFIG_SMP is explcitly not configured. In that test, the above logic would not work.

Reply:
The MP configuration is not the same as the SMP configuration. In the former there is no notion of multiple CPUs from Zephyr's perspective. Look at the `_current_cpu` definition for example. And obviously the kernel/mp test ddoes pass with this.